### ISSUE TITLE
Training gets Stuck on a Virtual Shared HDD enviroment at the end of the training.

### ISSUE THREAD

**Issue by lockder** [OpenNMT/OpenNMT-tf#206]
I found this happens me almost everytime I train(some times did work). 
Currently I'm using 4 machines
1 chief + ps
3 workers

all them have shared their files using a virtual hdd.

So usally the log end like this:

INFO:tensorflow:words_per_sec/features: 11394.3
INFO:tensorflow:loss = 0.0007880032, step = 6842 (393.713 sec)
INFO:tensorflow:loss = 0.00072970986, step = 6946 (396.764 sec)
INFO:tensorflow:loss = 0.00072739273, step = 6950 (400.848 sec)
INFO:tensorflow:Saving checkpoints for 7004 into /mnt/shared/NaturalLanguageRecognition/NaturalLanguageManager/src/Entity_Recognition/model/model.ckpt.
INFO:tensorflow:Loss for final step: 0.00024306774.
INFO:tensorflow:Loss for final step: 0.0007157028.
INFO:tensorflow:Loss for final step: 0.0007211566.

What I see is the chief machine report usually needs around 220sec to do 500 steps and report a summary and the workers need between 390 and 410 to do the 500 steps summary.

I tried to let the training end, but it never ends. I have to stop it manually, then running it again to to the avg last check points ( i tried also without the avg checkpoints and happens the same thing )

**Comment by guillaumekln** [comment]
If I understand correctly, the training is finished but some processes do not exit. Is that correct? If yes, which one in particular gets stuck?

**Comment by lockder** [comment]
yes, thats the issue. 
Sometimes the workers doesn't finish, but usually its the chief with the ps which doesn't finish.
I see the chief is going faster than the workers and may is creating the issue

**Comment by guillaumekln** [comment]
AFAIK this is a standard TensorFlow issue. Here is an answer from a TensorFlow developer:

> There's no built-in API for shutting down a TensorFlow server remotely; the assumption is that whatever mechanism (e.g. a cluster manager) you used to start the processes would also kill them after the job completes.

https://github.com/tensorflow/tensorflow/issues/4713

So while there are certainly ways to shut down everything from the code, the current design assumes a higher-level management system that does that.
